web_custom_label/models/ir_ui_view.py

```diff
@@ -0,0 +1,47 @@
+# © 2018 - today Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from lxml import etree
+from odoo import api, models
+from .common import set_custom_labels_on_fields
+
+
+class ViewWithCustomLabels(models.Model):
+
+    _inherit = 'ir.ui.view'
+
+    @api.model
+    def postprocess_and_fields(self, model, node, view_id):
+        """Add custom labels to the view xml.
+
+        This method is called in Odoo when generating the final xml of a view.
+        """
+        arch, fields = super().postprocess_and_fields(model, node, view_id)
+        lang = self.env.context.get('lang') or self.env.user.lang
+        labels = self.env['web.custom.label'].get(model, lang)
+        arch_with_custom_labels = _add_custom_labels_to_view_arch(labels, arch)
+        set_custom_labels_on_fields(labels, fields)
+        return arch_with_custom_labels, fields
+
+
+def _add_custom_labels_to_view_arch(labels, arch):
+    if not labels:
+        return arch
+
+    tree = etree.fromstring(arch)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Using lxml.etree.fromstring to parse untrusted XML data is known to be vulnerable to XML attacks. Replace lxml.etree.fromstring with its defusedxml equivalent function.](https://app.codacy.com/manual/numigi/odoo-web-addons/pullRequest?prid=4960433)